src/scsiencrypt.cpp

```diff
@@ -194,66 +194,61 @@ SCSI_PAGE_INQ *SCSIGetInquiry(std::string tapeDevice) {
   return status;
 }
 
-// Writes encryption options to the tape drive
-bool SCSIWriteEncryptOptions(std::string tapeDevice,
-                             SCSIEncryptOptions *eOptions) {
-
-  char buffer[1024];
-  memset(&buffer, 0, 1024);
-
+size_t SCSIInitSDEPage(SCSIEncryptOptions& eOptions,
+                       uint8_t *buffer) {
   SSP_PAGE_SDE options;
   // copy the template over the options
   memset(&options, 0, sizeof(SSP_PAGE_SDE));
   byteswap((unsigned char *)&options.pageCode, 2, 0x10);
-  int pagelen = sizeof(SSP_PAGE_SDE);
+  size_t pagelen = sizeof(SSP_PAGE_SDE);
   options.scope = 2; // all IT nexus = 10b
-  options.RDMC = eOptions->rdmc;
-  options.ckod = (eOptions->CKOD) ? 1 : 0;
+  options.RDMC = eOptions.rdmc;
```

Comment:
these rewrites should be part of a separate PR. Ideally one PR for adding tests only and one PR for each change.
Else we can introduce new mistakes quickly.

Reply:
I tend agree with @jonasstein.  If we could split the tests from the functionality changes that would be awesome.  Thanks for contributing this @jmwilson!